samples/smp/pktqueue/src/main.c

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "pktqueue.h"
+#include "main.h"
+
+static struct k_thread tthread[THREADS_NUM*QUEUE_NUM];
+static struct k_thread qthread[QUEUE_NUM];
+
+/* Each queue has its own mutex */
+struct k_mutex sender_queue_mtx[QUEUE_NUM];
+struct k_mutex receiver_queue_mtx[QUEUE_NUM];
+
+/* Variable which indicates the amount of processed queues */
+int queues_remain = QUEUE_NUM;
+/* Variable to define current queue in thread */
+int current_queue;
+
+/* Array of packet header descriptors */
+struct phdr_desc descriptors[QUEUE_NUM][SIZE_OF_QUEUE];
+
+/* Arrays of receiver and sender queues */
+struct phdr_desc_queue sender[QUEUE_NUM], receiver[QUEUE_NUM];
+
+/* Array of packet headers */
+uint8_t headers[QUEUE_NUM][SIZE_OF_QUEUE][SIZE_OF_HEADER];
+
+static K_THREAD_STACK_ARRAY_DEFINE(tstack, THREADS_NUM*QUEUE_NUM, STACK_SIZE);
+static K_THREAD_STACK_ARRAY_DEFINE(qstack, QUEUE_NUM, STACK_SIZE);
+
+K_MUTEX_DEFINE(fetch_queue_mtx);
+
+/* Function for initializing "sender" packet header queue */
+void init_datagram_queue(struct phdr_desc_queue *queue, int queue_num)
+{
+	queue->head = descriptors[queue_num];
+
+	for (int i = 0; i < SIZE_OF_QUEUE; i++) {
+		queue->tail = &descriptors[queue_num][i];
+		descriptors[queue_num][i].ptr = (uint8_t *)&headers[queue_num][i];
+		/* Fill packet header with random values */
+		for (int j = 0; j < SIZE_OF_HEADER; j++) {
+			/* leave crc field zeroed */
+			if (j < 10 || j > 11)
```

Comment:
I don't like these hard-coded 10 & 11 values across the code. I'd suggest to create human readable defines for them and even mention them in the documentation so it's easier to see where did they come from.